cli/cobra.go

```diff
@@ -46,21 +94,68 @@ func FlagErrorFunc(cmd *cobra.Command, err error) error {
 	}
 }
 
-var helpCommand = &cobra.Command{
-	Use:               "help [command]",
-	Short:             "Help about the command",
-	PersistentPreRun:  func(cmd *cobra.Command, args []string) {},
-	PersistentPostRun: func(cmd *cobra.Command, args []string) {},
-	RunE: func(c *cobra.Command, args []string) error {
-		cmd, args, e := c.Root().Find(args)
-		if cmd == nil || e != nil || len(args) > 0 {
-			return errors.Errorf("unknown help topic: %v", strings.Join(args, " "))
-		}
+// VisitAll will traverse all commands from the root.
+// This is different from the VisitAll of cobra.Command where only parents
+// are checked.
+func VisitAll(root *cobra.Command, fn func(*cobra.Command)) {
+	for _, cmd := range root.Commands() {
+		VisitAll(cmd, fn)
+	}
+	fn(root)
+}
 
-		helpFunc := cmd.HelpFunc()
-		helpFunc(cmd, args)
-		return nil
-	},
+// DisableFlagsInUseLine sets the DisableFlagsInUseLine flag on all
+// commands within the tree rooted at cmd.
+func DisableFlagsInUseLine(cmd *cobra.Command) {
+	VisitAll(cmd, func(ccmd *cobra.Command) {
+		// do not add a `[flags]` to the end of the usage line.
+		ccmd.DisableFlagsInUseLine = true
+	})
+}
+
+func makeHelpCommand(dockerCli command.Cli) *cobra.Command {
+	return &cobra.Command{
+		Use:               "help [command]",
+		Short:             "Help about the command",
+		PersistentPreRun:  func(cmd *cobra.Command, args []string) {},
+		PersistentPostRun: func(cmd *cobra.Command, args []string) {},
+		RunE: func(c *cobra.Command, args []string) error {
+			root := c.Root()
+			cmd, args, e := root.Find(args)
+			if cmd == nil || e != nil || len(args) > 0 {
+				// If the root is not a plugin root and we
+				// have a single arg (the command) then try
+				// checking if it is a plugin and if it is
+				// then call it.
+				if len(args) == 1 && root.Annotations[annotPluginRootCommand] != "true" {
```

Comment:
You can use the function declared just below:
```go
if len(args) == 1 && !isPlugin(root)
```

Reply:
Actually this went away entirely in the newest version.